rpcs3/Emu/scoped_progress_dialog.cpp

```diff
@@ -10,17 +10,41 @@ shared_ptr<std::string> progress_dialog_string_t::get_string_ptr() const noexcep
 	{
 		text.reset();
 
+		if (old_val.text_count == 0)
+		{
+			return text;
+		}
+
 		if (old_val.text_index != umax)
 		{
 			if (auto ptr = g_progr_text_queue[old_val.text_index].load())
 			{
 				text = ptr;
 			}
+			else
+			{
+				// Search for available text (until the time another thread sets index)
+				const u64 queue_size = g_progr_text_queue.size();
+
+				for (u64 i = queue_size - 1;; i--)
+				{
+					if (i == umax)
```

Comment:
Use this as exit condition of for loop. Then break in the scope below.
Return text after the for loop. Less spaghetti